arrow/api.py

```diff
@@ -4,14 +4,74 @@
 
 """
 
+from datetime import date, datetime
+from datetime import tzinfo as dt_tzinfo
+from time import struct_time
+from typing import Any, List, Optional, SupportsFloat, Tuple, Type, Union, overload
 
+from arrow.arrow import TZ_EXPR, Arrow
 from arrow.factory import ArrowFactory
 
 # internal default factory.
 _factory = ArrowFactory()
 
 
-def get(*args, **kwargs):
+@overload
+def get(
+    *,
+    locale: str = "en_us",
+    tzinfo: Optional[TZ_EXPR] = None,
+    normalize_whitespace: bool = False,
+) -> Arrow:
+    ...
+
+
+@overload
+def get(
+    __obj: Union[
+        Arrow,
+        datetime,
+        date,
+        struct_time,
+        dt_tzinfo,
+        SupportsFloat,
+        str,
+        Tuple[int, int, int],
+        None,
+    ],
+    *,
+    locale: str = "en_us",
+    tzinfo: Optional[TZ_EXPR] = None,
+    normalize_whitespace: bool = False,
+) -> Arrow:
+    ...
+
+
+@overload
+def get(
+    __arg1: Union[datetime, date],
+    __arg2: TZ_EXPR,
+    *,
+    locale: str = "en_us",
+    tzinfo: Optional[TZ_EXPR] = None,
+    normalize_whitespace: bool = False,
+) -> Arrow:
+    ...
+
+
+@overload
+def get(
+    __arg1: str,
+    __arg2: Union[str, List[str]],
+    *,
+    locale: str = "en_us",
+    tzinfo: Optional[TZ_EXPR] = None,
+    normalize_whitespace: bool = False,
+) -> Arrow:
```

Comment:
[Positional Only Argument](https://www.python.org/dev/peps/pep-0570/) is perfect for this. but is new on 3.8. So it would be good if you can use the syntax when we deprecate 3.7